sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -437,6 +450,14 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
                 return _currentActivity;
             }
 
+#if !NET5_0
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The values being passed into Write have the commonly used properties being preserved with DynamicDependency on the ActivityAdapter.Start() method.")]
```

Comment:
```suggestion
            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The values being passed into Write have the commonly used properties being preserved with DynamicDependency on the ActivityAdapter.Start() method, or the responsibility is on the user of this struct since the struct constructor is marked with RequiresUnreferencedCode.")]
```

To make the comment cover both the `_currentActivity` and the `_diagnosticSourceArgs` cases.